backend/clamav/ClamAvService.cs

```diff
@@ -0,0 +1,97 @@
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using nClam;
+using Pims.Av.Configuration;
+using Pims.Core.Exceptions;
+
+namespace Pims.Av
+{
+    /// <summary>
+    /// ClamAvService class, provides a service for integration with a running ClamAv Service and performing scans.
+    /// </summary>
+    public class ClamAvService : IAvService
+    {
+        private readonly ILogger<IAvService> _logger;
+
+        #region Constructors
+
+        /// <summary>
+        /// Creates a new instance of a ClamAvService, initializes with specified arguments.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="client"></param>
+        /// <param name="logger"></param>
+        public ClamAvService(IOptions<ClamAvOptions> options, IClamClient client, ILogger<IAvService> logger)
+        {
+            this.Options = options.Value;
+            this.Client = client;
+            this.Client.Port = this.Options.Port;
+            this.Client.Server = this.Options.HostUri;
+            this._logger = logger;
+        }
+        #endregion
+
+        #region Properties
+        public ClamAvOptions Options { get; }
+        protected IClamClient Client { get; }
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Scan the passed file, throw exceptions if the scan fails, or returns a positive result.
+        /// </summary>
+        /// <param name="file"></param>
+        /// <returns></returns>
+        public async Task Scan(IFormFile file)
+        {
+            if (this.Options.DisableScan)
+            {
+                this._logger.LogInformation("ClamAV scan disabled");
+                return;
+            }
+            using var ms = new MemoryStream();
+            file.OpenReadStream().CopyTo(ms);
+            byte[] fileBytes = ms.ToArray();
+
+            await Scan(fileBytes);
+        }
+
+        /// <summary>
+        /// Scan the passed file, throw exceptions if the scan fails, or returns a positive result.
+        /// </summary>
+        /// <param name="fileData"></param>
+        /// <returns></returns>
+        public async Task Scan(byte[] fileData)
+        {
+            if (this.Options.DisableScan)
+            {
+                this._logger.LogInformation("ClamAV scan disabled");
+                return;
+            }
+            this._logger.LogInformation("ClamAV scan start");
+            var scanResult = await this.Client.SendAndScanFileAsync(fileData);
+
+            switch (scanResult.Result)
+            {
+                case ClamScanResults.Unknown:
+                    this._logger.LogError("ClamAV scan failed", scanResult);
+                    throw new AvException("Virus scan failed with unknown response. If you continue to see this error, please contact your site administrator.");
```

Comment:
handle all failure cases with exceptions, these will be reported normally by the frontend using our existing mechanisms.